iceoryx2/src/port/listener.rs

```diff
@@ -32,65 +32,74 @@
 //!
 //! See also [`crate::port::listener::Listener`]
 
-use iceoryx2_bb_lock_free::mpmc::unique_index_set::UniqueIndex;
 use iceoryx2_bb_log::fail;
 use iceoryx2_cal::dynamic_storage::DynamicStorage;
 use iceoryx2_cal::event::{ListenerBuilder, ListenerWaitError};
 use iceoryx2_cal::named_concept::NamedConceptBuilder;
 
 use crate::service::naming_scheme::event_concept_name;
 use crate::{port::port_identifiers::UniqueListenerId, service};
-use std::{marker::PhantomData, time::Duration};
+use std::rc::Rc;
+use std::time::Duration;
 
 use super::event_id::EventId;
 use super::listen::{Listen, ListenerCreateError};
 
 /// Represents the receiving endpoint of an event based communication.
 #[derive(Debug)]
-pub struct Listener<'a, Service: service::Service> {
-    _dynamic_config_guard: Option<UniqueIndex<'a>>,
+pub struct Listener<Service: service::Service> {
+    dynamic_listener_handle: u32,
     listener: <Service::Event as iceoryx2_cal::event::Event<EventId>>::Listener,
     cache: Vec<EventId>,
-    _phantom_a: PhantomData<&'a Service>,
+    dynamic_storage: Rc<Service::DynamicStorage>,
 }
 
-impl<'a, Service: service::Service> Listener<'a, Service> {
-    pub(crate) fn new(service: &'a Service) -> Result<Self, ListenerCreateError> {
+impl<Service: service::Service> Drop for Listener<Service> {
+    fn drop(&mut self) {
+        self.dynamic_storage
+            .get()
+            .event()
+            .release_listener_id(self.dynamic_listener_handle)
```

Comment:
I think the name `release_listener_id` is misleading since it requires a different ID than what was used for `add_listener_id`

Reply:
Renamed it to `release_{**}_handle` for all the other occurrences.